web/cypress/views/incidents-page.ts

```diff
@@ -0,0 +1,130 @@
+  },
+
+  selectIncidentByBarIndex: (index = 0) => {
+    cy.log('incidentsPage.selectIncidentByBarIndex');
+    // TODO: There are two bars for the same incident, one transparent, one not. 
+    // Right now, we use both which works but doubles the execution time.
+    return cy.waitUntil(() => {
+      return incidentsPage.elements.incidentsChartCard()
+        .find('path[role="presentation"]')
+        .eq(index)
+        .click({ force: true })
+        .then(() => {
+          return Cypress.$('[aria-label="alerts-table"]').length > 0;
+        });
+    }, { interval: 10000, timeout: 120000 })
+    .then(() => incidentsPage.elements.alertsTable().scrollIntoView().should('be.visible'))
+  },
+
+  expandRow: (rowIndex = 0) => {
+    cy.log('incidentsPage.expandRow');
+    incidentsPage.elements.alertsTable()
+      .find('tbody')
+      .eq(rowIndex)
+      .within(() => {
+        cy.get('[aria-label="Details"], button[aria-expanded], button.pf-m-plain')
+          .first()
+          .click({ force: true });
+      });
+  },
+
+
+  findIncidentWithAlert: (alertName: string): Cypress.Chainable<boolean> => {
+    cy.log(`incidentsPage.findIncidentWithAlert: ${alertName}`);
+    return incidentsPage.elements.incidentsChartCard()
+      .find('path[role="presentation"]')
+      .then(($bars) => {
+        const totalBars = $bars.length;
+        if (totalBars <= 3) { // 3 paths are always present in the legend,
+        // their parent is g without presentation label opposed ot the proper bars
+          cy.task('log', 'No bars found in incidents chart');
+          return cy.wrap(false);
+        }
+
+        const tryIndex = (index: number): Cypress.Chainable<boolean> => {
+          if (index >= totalBars - 3) {
+            return cy.wrap(false);
+          }
+
+          return cy
+            .wrap(null)
+            .then(() => {
+              incidentsPage.selectIncidentByBarIndex(index);
+              return null;
+            })
+            .then(() => incidentsPage.elements.alertsTable().invoke('text'))
+            .then((text) => {
+              if (String(text).includes(alertName)) {
+                return cy.wrap(true);
+              }
+              // Expand a row if present to surface nested details
+              incidentsPage.expandRow(0);
+              return incidentsPage.elements.alertsTable().invoke('text').then((text2) => {
+                if (String(text2).includes(alertName)) {
+                  return cy.wrap(true);
+                }
+                return tryIndex(index + 1);
+              });
+            });
+        };
+
+        return tryIndex(0);
+      });
+  },
+
+  // Centralized element selectors - all selectors defined in one place
+  elements: {
+    // Page structure
+    toolbar: () => cy.get('#toolbar-with-filter'),
+
+    // Controls and filters
+    daysSelect: () => cy.get('button[data-ouia-component-id="OUIA-Generated-MenuToggle-4"]'),
```

Comment:
just to mention it seems a bit fragile, but if you tested and it always ends -4, go ahead. other way would be something like this: button[data-ouia-component-id^="OUIA-Generated-MenuToggle".

Reply:
@DavidRajnoha you could ask dev team to create a unique  id for this element like `[data-testid="days-select"]`

Reply:
I will focus on after this gets merged and make a PR that adds the selectors to the UI and uses them in the tests.